src/bin/huak/commands/config/completion.rs

```diff
@@ -0,0 +1,63 @@
+use crate::commands::Cli;
+use crate::errors::CliResult;
+
+use huak::ops::config;
+
+use clap::{Args, Command, CommandFactory, Subcommand};
+use clap_complete::{generate, Shell};
+
+/// Prints the script to stdout and a way to add the script to the shell init file to stderr. This
+/// way if the user runs completion <shell> > completion.sh only the stdout will be redirected into
+/// completion.sh.
+pub fn run(config_command: Config) -> CliResult<()> {
+    match config_command.command {
+        ConfigCommand::Completion { shell } => {
+            generate_shell_completion_script(shell)
+        }
+        ConfigCommand::Install { shell } => {
+            let mut cmd: Command = Cli::command();
+            let _result = match shell {
+                Shell::Bash => config::_add_completion_bash(),
+                Shell::Elvish => config::_add_completion_elvish(),
+                Shell::Fish => config::_add_completion_fish(&mut cmd),
+                Shell::PowerShell => config::_add_completion_powershell(),
+                Shell::Zsh => config::_add_completion_zsh(&mut cmd),
+                _ => Ok(()),
+            };
+        }
+        ConfigCommand::Uninstall { shell } => {
+            let _result = match shell {
+                Shell::Bash => config::_remove_completion_bash(),
+                Shell::Elvish => config::_remove_completion_elvish(),
+                Shell::Fish => config::_remove_completion_fish(),
+                Shell::PowerShell => config::_remove_completion_powershell(),
+                Shell::Zsh => config::_remove_completion_zsh(),
```

Comment:
AFAIK (newish to rust) prefixing functions with an underscore is used to invoke `#[allow(dead_code)]` on functions. So I'm not sure it's idiomatic to do this.

Reply:
Also `huak::ops` is meant for operation implementations for `Project` and related contexts. This seems more like a CLI app feature. So we can throw those back in here.

Reply:
> AFAIK (newish to rust) prefixing functions with an underscore is used to invoke #[allow(dead_code)] on functions. So I'm not sure it's idiomatic to do this.

Yes I initially did this to hide some noise during testing, I'll remove this.

> Also huak::ops is meant for operation implementations for Project and related contexts. This seems more like a CLI app feature. So we can throw those back in here.

I have some test declarations in my code to inject a different test file. For some weird reason these did not take so I thought there was some weirdness going on with the lib and bin being separate. Turns out that's not the case after some more testing so I'll move them back to the bin.